test/functional/plugin/lsp/diagnostic_spec.lua

```diff
@@ -400,7 +400,9 @@ describe('vim.lsp.diagnostic', function()
       )
 
       exec_lua(function()
-        vim.lsp.stop_client(client_id2)
+        if client_id2 ~= nil then
```

Comment:
I'm not sure why this one needed a nil-check, but the test breaks without it. Perhaps this uncovered a bug in the test that `vim.lsp.stop_client()` was accidentally covering up by accepting `nil` as an argument?

Reply:
Yeah, making it a global variable fixed it. Thanks for the suggestion.